### PR TITLE
VCC Glitch Mosfet Control Update

### PR DESCRIPTION
**Pull request by DJShepherd**
* Refactored VCC glitch mosfet control
* Refactored default and glitch setup/disable API
* Added message utilities; Added optimized data copy
* Added API utilities

Passed tests/api_test.py, validated glitch_hp/lp value assignment and readback.

### CODE REVIEW

**Comment by DJShepherd** [comment]
A big chunk of changes in ChipWhispererExtra.py is because I moved CWExtraSettings above GPIOSettings so I could access the CWExtraSettings class globals from GPIOSettings globals.

I would test the CWLITE VCC glitch demos but I think a long time ago, I burned out my mosfets cuz I can't hit glitches anymore :( I bought some replacement mosfets, but I have to find time to do the soldering work on the board.

I do have a HUSKY, but I haven't had time to set it up.  I do plan on playing around with it soon and use that as my new testing/development platform.

**Comment by DJShepherd** [comment]
To see more of the direction I'm proposing for the CONSTS/API interface that I created in this PR, I'm currently working on the same interface for the TIO/GPIO API as seen here:
[Compare: DJShepherd/fast_glitcht...DJShepherd/gpio_api_update](https://github.com/DJShepherd/chipwhisperer/compare/DJShepherd/fast_glitcht...DJShepherd:chipwhisperer:DJShepherd/gpio_api_update)

**Comment by DJShepherd** [comment]
Any progress on getting this validated and merged in? I've been keeping it updated and making sure nothing breaks in terms of the workflow I have been testing with.

**Comment by alex-dewar** [comment]
Sorry about the delays with this. I've been working on stuff related to an upcoming training, so I don't want to make too many big changes before that.

**Comment by DJShepherd** [comment]
> Sorry about the delays with this. I've been working on stuff related to an upcoming training, so I don't want to make too many big changes before that.

Understandable, thanks for the update, was just curious! :)  I'll just keep working on the other changes slowly, I've been busy as of late, too.

File: software/chipwhisperer/capture/scopes/OpenADC.py [hunk 1]
```diff
@@ -322,39 +325,22 @@ def default_setup(self, verbose=True):
...
         if self._is_husky:
-            self.clock.clkgen_src = 'system'
-            self.clock.clkgen_freq = 7.37e6
-            self.clock.adc_mul = 4
-            while not self.clock.clkgen_locked:
-                count += 1
-                self.clock.reset_dcms()
-                if count > 10:
-                    raise OSError("Could not lock PLL. Try rerunning this function or calling scope.pll.reset(): {}".format(self))
+            if not self.clock.try_wait_clkgen_locked(10):
```
Comment (alex-dewar) [inline software/chipwhisperer/capture/scopes/OpenADC.py]: This needs a Husky version as well (`cwhardware/ChipWhispererHuskyClock.py`)

File: software/chipwhisperer/capture/scopes/_OpenADCInterface.py [hunk 2]
```diff
@@ -452,6 +553,9 @@ def lo_gain_errors_disabled(self):
         else:
             return False
 
+    def disable_clip_and_lo_gain_errors(self, disabled):
```
Comment (alex-dewar) [inline software/chipwhisperer/capture/scopes/_OpenADCInterface.py]: This needs to be in `scope.errors` instead of `scope.adc`.

Comment (alex-dewar) [reply software/chipwhisperer/capture/scopes/_OpenADCInterface.py]: Whoops, this is just in the wrong class in this file (needs to be in `TriggerSettings`)

**Comment by alex-dewar** [comment]
Will run this on a few more scope/target setups, but should be good once the fixes above are implemented

**Comment by DJShepherd** [comment]
I've been a bit busy so I haven't gotten to it, but it's on my radar and I'm hoping to tackle it this weekend!

File: software/chipwhisperer/capture/scopes/_OpenADCInterface.py (hunk 2 above)
Comment (DJShepherd) [reply software/chipwhisperer/capture/scopes/_OpenADCInterface.py]: Fixed.

File: software/chipwhisperer/capture/scopes/OpenADC.py (hunk 1 above)
Comment (DJShepherd) [reply software/chipwhisperer/capture/scopes/OpenADC.py]: This is common logic so for now, in order to not duplicate/copy-paste code, I just moved it to be an OpenADC method with a note to move it into a base/common clock class if refactoring ever happens.

**Comment by alex-dewar** [comment]
Can confirm the latest update works. Will merge in now.

**Pull request merged**